### DISCUSSION TITLE
Clarity on alternate DB support?

### DISCUSSION THREAD

**Discussion by ENBYSS** [pocketbase/pocketbase#1395] [Q&A] [answered]
Mostly wondering since I've seen some conflicting information. Feel free to let me know if it's not certain right now.

On the website's [FAQ](https://pocketbase.io/faq) it says that there are _no_ plans for supporting other databases.

However in the project [roadmap for **v1.0.0**](https://github.com/orgs/pocketbase/projects/2/views/1), I can see that at the bottom of the **Todo** list there's `Support external databases (MySQL, MariaDB, PostgreSQL)`.

Basically, I'm wondering which of these are accurate. Or alternatively, whether there's doubt surrounding this feature (as an example, it needs more research to see how complicated/worth it it is).

If the roadmap is correct - then I have a further question - being whether supporting Redis could be possible in the future. Not necessarily v1.0.0 - but _at some point_. Mostly because of how PB allows for realtime connections, it would work incredibly well with a Redis instance, updating the cache on modification/addition/deletion.

That last question is more flippant than anything - been looking into Redis and part of me considered using it, but PB is already fast enough. Just wanna be as performant as possible due to scaling.

**Comment by ganigeorgiev** [comment, marked as answer]
Support for other databases was added to the roadmap after the discussion in https://github.com/pocketbase/pocketbase/discussions/14#discussioncomment-3109447, but for now this has the lowest priority and it is there just as a reminder, meaning that it may be not added with the final v1.
The main reason for now to even consider adding support for other SQL databases is to support cloud providers with ephemeral storage, not because of scaling (there are other things beside the database that prevents PocketBase to scale horizontally and this is by design).

There are no plans for Redis, since to maximize the performance, we rely heavily on the SQL syntax (even the API rules are converted to SQL statements).

**Reply by ganigeorgiev** [reply]
I've removed the task from the roadmap and added it to my internal todo to avoid future confusions.

**Reply by ENBYSS** [reply]
Thanks for the elaboration! Regarding horizontal scaling, I feel as if a project on _that_ scale would need its own specialized/custom backend anyways, at which point a migration from PB should prove uncomplicated. For 99.99% of projects, it's unneeded, so it's not a worry of mine heh.

Didn't know there was a point regarding ephemeral storage, that's pretty interesting!

**Comment by Ohene-Asase** [comment]
why did i receive this email?

**Reply by ganigeorgiev** [reply]
@Ohene-Asase Sorry, I don't have control over the notifications. It is some spam bot that tagged several users. GitHub already disabled their account(s).
